src/EditorFeatures/Core/Implementation/NavigateTo/NavigateToItemProvider.Searcher.cs

```diff
@@ -30,15 +32,24 @@ private class Searcher
                 ItemDisplayFactory displayFactory,
                 INavigateToCallback callback,
                 string searchPattern,
+                bool searchCurrentDocument,
                 CancellationToken cancellationToken)
             {
                 _solution = solution;
                 _displayFactory = displayFactory;
                 _callback = callback;
                 _searchPattern = searchPattern;
+                _searchCurrentDocument = searchCurrentDocument;
                 _cancellationToken = cancellationToken;
                 _progress = new ProgressTracker(callback.ReportProgress);
                 _asyncListener = asyncListener;
+
+                if (_searchCurrentDocument)
```

Comment:
I'm split 50/50 on whether to have this pattern, or just pass the document in, and have non-null indicate search only in that document...